miio/cooker.py

```diff
@@ -0,0 +1,706 @@
+                MODEL_NORMAL5]
+
+MODEL_NORMAL_GROUP1 = [MODEL_NORMAL2, MODEL_NORMAL5]
+MODEL_NRRMAL_GROUP2 = [MODEL_NORMAL3, MODEL_NORMAL4]
+
+COOKING_STAGE_NAME = [
+    'Preheating',
+    'Water-absorbing',
+    'Boiling',
+    'Gelantinizing',
+    'Braising',
+]
+
+COOKING_STAGE_DESCRIPTION = [
+    'Starting heating, to soften rice gradually',
+    'Increase temperature, to flesh grains with water',
+    'Last high heating, to cook rice evenly',
+    'Steaming under high temperature, to bring sweetness to grains',
+    'Reheat the rice to open its taste',
+]
+
+
+class CookerException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    # Observed
+    Running = 'running'
+    Waiting = 'waiting'
+    AutoKeepWarm = 'autokeepwarm'
+    # Potential candidates
+    Cooking = 'cooking'
+    Finish = 'finish'
+    FinishA = 'finisha'
+    KeepWarm = 'keepwarm'
+    KeepTemp = 'keep_temp'
+    Notice = 'notice'
+    Offline = 'offline'
+    Online = 'online'
+    PreCook = 'precook'
+    Resume = 'resume'
+    ResumeP = 'resumep'
+    Start = 'start'
+    StartP = 'startp'
+    Cancel = 'Отмена'
+
+
+class TemperatureHistory:
+    def __init__(self, data: str):
+        """
+        Example values:
+
+        2 minutes:
+        161515161c242a3031302f2eaa2f2f2e2f
+
+        12 minutes:
+        161515161c242a3031302f2eaa2f2f2e2f2e302f2e2d302f2f2e2f2f2f2f343a3f3f3d3e3c3d3c3f3d3d3d3f3d3d3d3d3e3d3e3c3f3f3d3e3d3e3e3d3f3d3c3e3d3d3e3d3f3e3d3f3e3d3c
+
+        32 minutes:
+        161515161c242a3031302f2eaa2f2f2e2f2e302f2e2d302f2f2e2f2f2f2f343a3f3f3d3e3c3d3c3f3d3d3d3f3d3d3d3d3e3d3e3c3f3f3d3e3d3e3e3d3f3d3c3e3d3d3e3d3f3e3d3f3e3d3c3f3e3d3c3f3e3d3c3f3f3d3d3e3d3d3f3f3d3d3f3f3e3d3d3d3e3e3d3daa3f3f3f3f3f414446474a4e53575e5c5c5b59585755555353545454555554555555565656575757575858585859595b5b5c5c5c5c5d5daa5d5e5f5f606061
+
+        55 minutes:
+        161515161c242a3031302f2eaa2f2f2e2f2e302f2e2d302f2f2e2f2f2f2f343a3f3f3d3e3c3d3c3f3d3d3d3f3d3d3d3d3e3d3e3c3f3f3d3e3d3e3e3d3f3d3c3e3d3d3e3d3f3e3d3f3e3d3c3f3e3d3c3f3e3d3c3f3f3d3d3e3d3d3f3f3d3d3f3f3e3d3d3d3e3e3d3daa3f3f3f3f3f414446474a4e53575e5c5c5b59585755555353545454555554555555565656575757575858585859595b5b5c5c5c5c5d5daa5d5e5f5f60606161616162626263636363646464646464646464646464646464646464646364646464646464646464646464646464646464646464646464646464aa5a59585756555554545453535352525252525151515151
+        """
+        self.data = [int(data[i:i + 2], 16) for i in range(0, len(data), 2)]
+
+    @property
+    def temperatures(self) -> list[int]:
+        return self.data
+
+    def __str__(self) -> str:
+        return "".join(["{:02x}".format(value) for value in self.data])
+
+    def __repr__(self) -> str:
+        s = "<TemperatureHistory temperatures=%s>" % self.data
+        return s
+
+
+class CookerCustomizations:
+    def __init__(self, custom: str):
+        """
+        Example values:
+
+        ffffffffffff011effff010000001d1f,
```

Comment:
It would be nice to have some sort of short description for the bytes / how to interpret these.